rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -58,16 +65,64 @@ public LRARecoveryModule(LRAService lraService) {
         RecordTypeManager.manager().add(new RecordTypeMap() {
             @Override
             public Class<? extends AbstractRecord> getRecordClass() {
-                return null;
+                return LRAParticipantRecord.class;
             }
 
             @Override
             public int getType() {
-                return 0;
+                return RecordType.LRA_RECORD;
+            }
+        });
+        RecordTypeManager.manager().add(new RecordTypeMap() {
+            @Override
+            public Class<? extends AbstractRecord> getRecordClass() {
+                return LRAParentAbstractRecord.class;
+            }
+
+            @Override
+            public int getType() {
+                return RecordType.LRA_PARENT_RECORD;
+            }
+        });
+        RecordTypeManager.manager().add(new RecordTypeMap() {
+            @Override
+            public Class<? extends AbstractRecord> getRecordClass() {
+                return LRAChildAbstractRecord.class;
+            }
+
+            @Override
+            public int getType() {
+                return RecordType.LRA_CHILD_RECORD;
             }
         });
     }
 
+    private static LRAService service;
+    public static LRAService getService() {
+        if (service == null) {
```

Comment:
okay

Reply:
The routine is re-entrant so no need for any double check idiom.
That said I have reworked the code that initialises and accesses the singleton and I have introduced the double check. I'll push it when we have some agreement on the other concerns.

Reply:
I decided that the double check isn't needed after all and since synchronisation is an expensive operation I have not included it. I added some commentary to the routine to explain why it's not neccessary.